backend/uclapi/roombookings/helpers.py

```diff
@@ -230,3 +248,60 @@ def how_many_seconds_until_midnight():
         day=tomorrow.day, hour=0, minute=0, second=0
     )
     return (midnight - datetime.datetime.now()).seconds
+
+
+def _get_bookings_for_room(bookings, siteid, roomid):
```

Comment:
Can we not do this with a database query? It should be far more efficient (e.g. use the cache lock then do `bookings = BookingX.objects.filter(siteid=siteid, roomid=roomid)` (where X = A or B depending on the current state of the lock) then use our existing code to turn this into JSON. It should be far more efficient...